Query extension now uses expression instead of locator

sbonebrake opened the pull request.

maudineormsby commented:
Ah, interesting.

I don't have time to test this out but I checked the Capybara source and I think this is fine.

sbonebrake commented:
Yeah, it was an oversight I made in the first version because all of the locators pass through unchanged.  I'm testing this locally so I can add new selectors such as:
```ruby
# Call with both class and text: :complex_exact, "android.widget.TextView:::Welcome"
Capybara.add_selector(:complex_exact) do
  klass = locator.split(":::")[0]
  text = locator.split(":::")[1]
  custom(:uiautomator) { |locator|    Capybara.current_session.driver.appium_driver.string_visible_exact(klass, text) }
end
```
I think once I have those finalized I'll have another PR to add them here.

bootstraponline commented:
> Unrecognized author. If this is you, make sure the email address you used to commit is associated with your account. You can add emails to your account in Settings.

The code looks good. :+1:
